store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -353,29 +387,22 @@ protected boolean isColumnQualifierBytesValid(final byte[] bytes, final int numP
 
     @Override
     public long buildTimestamp(final String group, final Properties properties) {
-        Long timestamp;
+        Long timestamp = LongUtil.getTimeBasedRandom();
```

Comment:
I think the ordering can be changed so this doesn't happen first, given it will be a tiny bit expensive. Currently, `timeSensitiveAggregatedGroups` will have to call `System.currentTimeMillis()` twice.

Reply:
True good point.